isobus/src/isobus_time_date_interface.cpp

```diff
@@ -0,0 +1,218 @@
+	{
+		bool retVal = false;
+
+		if (nullptr != requestingControlFunction)
+		{
+			retVal = ParameterGroupNumberRequestProtocol::request_parameter_group_number(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TimeDate),
+			                                                                             requestingControlFunction,
+			                                                                             optionalDestination);
+		}
+		return retVal;
+	}
+
+	std::shared_ptr<InternalControlFunction> TimeDateInterface::get_control_function() const
+	{
+		return myControlFunction;
+	}
+
+	void TimeDateInterface::process_rx_message(const CANMessage &message, void *parentPointer)
+	{
+		auto timeDateInterface = static_cast<TimeDateInterface *>(parentPointer);
+
+		if ((nullptr != timeDateInterface) &&
+		    (static_cast<std::uint32_t>(CANLibParameterGroupNumber::TimeDate) == message.get_identifier().get_parameter_group_number()) &&
+		    (nullptr != message.get_source_control_function()))
+		{
+			if (CAN_DATA_LENGTH == message.get_data_length())
+			{
+				TimeAndDateInformation timeAndDateInformation;
+
+				timeAndDateInformation.controlFunction = message.get_source_control_function();
+				timeAndDateInformation.timeAndDate.seconds = message.get_uint8_at(0) / 4; // This is SPN 959
+				timeAndDateInformation.timeAndDate.milliseconds = (message.get_uint8_at(0) % 4) * 250; // This is also part of SPN 959
+				timeAndDateInformation.timeAndDate.minutes = message.get_uint8_at(1); // This is SPN 960
+				timeAndDateInformation.timeAndDate.hours = message.get_uint8_at(2); // This is SPN 961
+				timeAndDateInformation.timeAndDate.month = message.get_uint8_at(3); // This is SPN 963
+				timeAndDateInformation.timeAndDate.day = message.get_uint8_at(4) / 4; // This is SPN 962
+				timeAndDateInformation.timeAndDate.quarterDays = message.get_uint8_at(4) % 4; // This is also part of SPN 962
+				timeAndDateInformation.timeAndDate.year = static_cast<std::uint16_t>(message.get_uint8_at(5) + 1985); // This is SPN 964
+				timeAndDateInformation.timeAndDate.localMinuteOffset = static_cast<std::int8_t>(message.get_uint8_at(6) - 125); // This is SPN 1601
+				timeAndDateInformation.timeAndDate.localHourOffset = static_cast<std::int8_t>(message.get_int8_at(7) - 125); // This is SPN 1602
+
+#ifndef DISABLE_CAN_STACK_LOGGER
+				if (CANStackLogger::get_log_level() == CANStackLogger::LoggingLevel::Debug) // This is a very heavy log statement, so only do it if we are logging at debug level
+				{
+					std::ostringstream oss;
+					oss << "[Time/Date]: Control Function 0x";
+					oss << std::setfill('0') << std::setw(16) << std::hex << message.get_source_control_function()->get_NAME().get_full_name();
+					oss << " at address 0x" << std::setfill('0') << std::setw(2) << std::hex << static_cast<int>(message.get_source_control_function()->get_address());
```

Comment:
nit(non-blocking): I feel like it's more in line with the rest of the stack to use decimals for the address of a CF